storage/range_test.go

```diff
@@ -750,6 +750,67 @@ func TestRangeGossipConfigUpdates(t *testing.T) {
 	}
 }
 
+// TestRangeNoGossipConfig verifies that certain commnads (e.g.,
+// reads, writes in uncommitted transactions) do not trigger gossip.
+func TestRangeNoGossipConfig(t *testing.T) {
+	defer leaktest.AfterTest(t)
+	tc := testContext{}
+	tc.Start(t)
+	defer tc.Stop()
+
+	// Add a permission for a new key prefix.
+	db1Perm := &proto.PermConfig{
+		Read:  []string{"spencer"},
+		Write: []string{"spencer"},
+	}
+	key := keys.MakeKey(keys.ConfigPermissionPrefix, proto.Key("/db1"))
+	raftID := proto.RaftID(1)
+
+	txn := newTransaction("test", key, 1 /* userPriority */, proto.SERIALIZABLE, tc.clock)
+	data, err := gogoproto.Marshal(db1Perm)
+	if err != nil {
+		t.Fatal(err)
+	}
+	req1, reply1 := putArgs(key, data, raftID, tc.store.StoreID())
+	req1.Txn = txn
+	req1.Timestamp = txn.Timestamp
+
+	req2, reply2 := endTxnArgs(txn, true /* commit */, raftID, tc.store.StoreID())
+	req2.Timestamp = txn.Timestamp
+
+	// Do inconsistent read. Otherwise, a get will fail due to an
```

Comment:
We need to execute these commands through the store instead of the range so that we can retry on write intents.